src/workerd/api/actor-state.c++

```diff
@@ -492,11 +493,19 @@ jsg::Promise<void> DurableObjectStorage::deleteAll(
 
   auto deleteAll = cache->deleteAll(options);
 
-  auto& context = IoContext::current();
-  context.addTask(deleteAll.count.then([&metrics = currentActorMetrics()](uint deleted) {
+  const auto updateStorageDeletes = [&](IoContext& context,
+                                        ActorObserver& metrics,
+                                        kj::Promise<uint> promise) -> kj::Promise<void> {
+    // The ActorObserver& reference here is guaranteed to outlive this task, so
+    // accessing it after the co_await here is safe.
+    auto deleted = co_await promise;
```

Comment:
Just as an fyi, we've been trying to avoid making lambda coroutines. It should always be fine without captures and it happens to be fine with `this` captures, but the temptation to add actual value captures (which do not get persisted) is always there :).

Reply:
Yeah, I agree. The approach I'm wanting to take is to largely convert any lambdas that are used into non-capturing alternatives that can, as a next step, be moved away as much as possible from lambdas